tests/kani/Intrinsics/Math/fabsf32.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Check that `fabsf32` returns the expected results: absolute value if argument
+// is not NaN, otherwise NaN
+#![feature(core_intrinsics)]
+
+#[kani::proof]
+fn test_abs_finite() {
+    let x: f32 = kani::any();
+    kani::assume(!x.is_nan());
+    let abs_x = unsafe { std::intrinsics::fabsf32(x) };
+    if x < 0.0 {
+        assert!(-x == abs_x);
```

Comment:
At first I was surprised that this was `==` and not `diff < epsilon` but I see that floating point has an explicit sign bit https://en.wikipedia.org/wiki/Floating-point_arithmetic#IEEE_754:_floating_point_in_modern_computers